## DSC 640 Term Project
* Milestone 1
* Mallory Young

## Exploring the Data
* Importing code and packages needed for this project
* Importing and displaying the data frame
* Filtering the data to only include Florida Data
* Checking for missing values
* Filtering the data to only include the columns relevant to my story

In [1]:
import pandas as pd

In [2]:
#load the Excel file into a DataFrame
df = pd.read_excel('nationaldatabaseofchildcareprices.xlsx')

#display the DataFrame
df.head()

,State_Name,State_Abbreviation,County_Name,County_FIPS_Code,StudyYear,UNR_16,FUNR_16,MUNR_16,UNR_20to64,FUNR_20to64,...,MFCCToddler,MFCCToddler_flag,MFCCPreschool,MFCCPreschool_flag,_75FCCInfant,_75FCCInfant_flag,_75FCCToddler,_75FCCToddler_flag,_75FCCPreschool,_75FCCPreschool_flag
0,Alabama,AL,Autauga County,1001,2008,5.42,4.41,6.32,4.6,3.5,...,83.45,3.0,81.40,1.0,97.4,1.0,97.4,3.0,95.0,1.0
1,Alabama,AL,Autauga County,1001,2009,5.93,5.72,6.11,4.8,4.6,...,87.39,3.0,85.68,1.0,102.0,1.0,102.0,3.0,100.0,1.0
2,Alabama,AL,Autauga County,1001,2010,6.21,5.57,6.78,5.1,4.6,...,91.33,3.0,89.96,1.0,106.6,1.0,106.6,3.0,105.0,1.0
3,Alabama,AL,Autauga County,1001,2011,7.55,8.13,7.03,6.2,6.3,...,95.28,3.0,94.25,1.0,111.2,1.0,111.2,3.0,110.0,1.0
4,Alabama,AL,Autauga County,1001,2012,8.60,8.88,8.29,6.7,6.4,...,99.22,3.0,98.53,1.0,115.8,1.0,115.8,3.0,115.0,1.0


In [3]:
df.shape

(34567, 227)

In [4]:
#filter the DataFrame to keep only rows where State_Name is Florida
florida_df = df[df['State_Name'] == 'Florida']

#display the filtered DataFrame
florida_df.head(50)

,State_Name,State_Abbreviation,County_Name,County_FIPS_Code,StudyYear,UNR_16,FUNR_16,MUNR_16,UNR_20to64,FUNR_20to64,...,MFCCToddler,MFCCToddler_flag,MFCCPreschool,MFCCPreschool_flag,_75FCCInfant,_75FCCInfant_flag,_75FCCToddler,_75FCCToddler_flag,_75FCCPreschool,_75FCCPreschool_flag
3519,Florida,FL,Alachua County,12001,2008,5.18,5.50,4.88,4.10,4.70,...,104.96,1.0,102.82,3.0,125.00,1.0,122.50,1.0,120.00,3.0
3520,Florida,FL,Alachua County,12001,2009,6.75,6.45,7.03,5.80,5.60,...,107.10,1.0,102.82,3.0,125.00,1.0,125.00,1.0,120.00,3.0
3521,Florida,FL,Alachua County,12001,2010,6.69,6.75,6.64,5.80,6.10,...,111.57,1.0,102.94,3.0,131.25,1.0,130.47,1.0,122.19,3.0
3522,Florida,FL,Alachua County,12001,2011,7.33,7.25,7.41,6.70,6.80,...,116.05,1.0,103.07,3.0,137.50,1.0,135.94,1.0,124.38,3.0
3523,Florida,FL,Alachua County,12001,2012,7.87,7.36,8.36,7.30,6.90,...,120.52,1.0,103.20,3.0,143.75,1.0,141.41,1.0,126.56,3.0
3524,Florida,FL,Alachua County,12001,2013,8.64,7.91,9.37,8.10,7.50,...,125.00,1.0,103.33,1.0,150.00,1.0,146.88,1.0,128.75,1.0
3525,Florida,FL,Alachua County,12001,2014,8.50,7.94,9.14,8.10,7.70,...,125.00,1.0,109.17,1.0,158.75,1.0,147.82,1.0,129.17,1.0
3526,Florida,FL,Alachua County,12001,2015,7.86,7.30,8.42,7.40,6.80,...,125.00,1.0,115.00,1.0,167.50,1.0,148.75,1.0,129.58,1.0
3527,Florida,FL,Alachua County,12001,2016,7.02,6.33,7.71,6.50,5.70,...,125.00,1.0,120.84,1.0,176.25,1.0,149.69,1.0,130.00,1.0
3528,Florida,FL,Alachua County,12001,2017,6.65,6.17,7.13,6.20,5.60,...,125.00,1.0,126.67,1.0,185.00,1.0,150.62,1.0,142.73,1.0


In [5]:
florida_df.shape

(737, 227)

In [6]:
#show all columns
pd.set_option('display.max_columns', None) 
#show all rows
pd.set_option('display.max_rows', None) 
#prevent wrapping
pd.set_option('display.expand_frame_repr', False)  
#creating a list of the column names
florida_df.columns.tolist()  
#I will use this list of column names to filter only the columns I want to keep

['State_Name',
 'State_Abbreviation',
 'County_Name',
 'County_FIPS_Code',
 'StudyYear',
 'UNR_16',
 'FUNR_16',
 'MUNR_16',
 'UNR_20to64',
 'FUNR_20to64',
 'MUNR_20to64',
 'FLFPR_20to64',
 'FLFPR_20to64_Under6',
 'FLFPR_20to64_6to17',
 'FLFPR_20to64_Under6_6to17',
 'MLFPR_20to64',
 'PR_F',
 'PR_P',
 'MHI',
 'ME',
 'FME',
 'MME',
 'MHI_2018',
 'ME_2018',
 'FME_2018',
 'MME_2018',
 'TotalPop',
 'OneRace',
 'OneRace_W',
 'OneRace_B',
 'OneRace_I',
 'OneRace_A',
 'OneRace_H',
 'OneRace_Other',
 'TwoRaces',
 'Hispanic',
 'Households',
 'H_Under6_BothWork',
 'H_Under6_FWork',
 'H_Under6_MWork',
 'H_Under6_SingleM',
 'H_6to17_BothWork',
 'H_6to17_Fwork',
 'H_6to17_Mwork',
 'H_6to17_SingleM',
 'EMP_M',
 'MEMP_M',
 'FEMP_M',
 'EMP_Service',
 'MEMP_Service',
 'FEMP_Service',
 'EMP_Sales',
 'MEMP_Sales',
 'FEMP_Sales',
 'EMP_N',
 'MEMP_N',
 'FEMP_N',
 'EMP_P',
 'MEMP_P',
 'FEMP_P',
 'iUNR_16',
 'iFUNR_16',
 'iMUNR_16',
 'iUNR_20to64',
 'iFUNR_20to64',
 'iMUNR_20to64',
 'iFLFPR_20to64',
 'iFLFPR_2

In [7]:
#finding the count of NaN values in each column
nan_counts = florida_df.isna().sum()

#Display the counts of NaN values for each column
print(nan_counts)

State_Name                      0
State_Abbreviation              0
County_Name                     0
County_FIPS_Code                0
StudyYear                       0
UNR_16                          0
FUNR_16                         0
MUNR_16                         0
UNR_20to64                      0
FUNR_20to64                     0
MUNR_20to64                     0
FLFPR_20to64                    0
FLFPR_20to64_Under6             0
FLFPR_20to64_6to17              0
FLFPR_20to64_Under6_6to17       0
MLFPR_20to64                    0
PR_F                            0
PR_P                            0
MHI                             0
ME                              0
FME                             0
MME                             0
MHI_2018                        0
ME_2018                         0
FME_2018                        0
MME_2018                        0
TotalPop                        0
OneRace                         0
OneRace_W                       0
OneRace_B     

In [8]:
#list of columns to keep (relevant to my story)
columns_to_keep = ['State_Name', 'County_Name', 'County_FIPS_Code', 'StudyYear', 'UNR_20to64', 'FUNR_20to64', 'MUNR_20to64', 'FLFPR_20to64_Under6',
                   'TotalPop', 'Households', 'H_Under6_BothWork', 'H_Under6_FWork', 'H_Under6_MWork', 'H_Under6_SingleM', 'MCInfant', 'MCInfant_flag', 
                   'MCToddler', 'MCToddler_flag', 'MCPreschool', 'MCPreschool_flag', '_75CInfant', '_75CInfant_flag', '_75CToddler', 
                   '_75CToddler_flag', '_75CPreschool', '_75CPreschool_flag']

#filter the DataFrame 
florida_df = florida_df[columns_to_keep]

In [9]:
#printing the first 20 rows of the dataframe
florida_df.head(20)

,State_Name,County_Name,County_FIPS_Code,StudyYear,UNR_20to64,FUNR_20to64,MUNR_20to64,FLFPR_20to64_Under6,TotalPop,Households,H_Under6_BothWork,H_Under6_FWork,H_Under6_MWork,H_Under6_SingleM,MCInfant,MCInfant_flag,MCToddler,MCToddler_flag,MCPreschool,MCPreschool_flag,_75CInfant,_75CInfant_flag,_75CToddler,_75CToddler_flag,_75CPreschool,_75CPreschool_flag
3519,Florida,Alachua County,12001,2008,4.10,4.70,3.60,67.10,239046,96894,5618,3675,498,4499.0,137.65,3.0,115.54,1.0,106.36,3.0,165.00,3.0,138.50,1.0,127.50,3.0
3520,Florida,Alachua County,12001,2009,5.80,5.60,6.00,67.70,238990,96518,5236,4026,398,4258.0,145.99,3.0,126.80,1.0,116.79,3.0,175.00,3.0,152.00,1.0,140.00,3.0
3521,Florida,Alachua County,12001,2010,5.80,6.10,5.60,67.30,244247,97540,5128,3696,431,4298.0,149.55,3.0,129.75,1.0,119.46,3.0,179.81,3.0,154.83,1.0,142.96,3.0
3522,Florida,Alachua County,12001,2011,6.70,6.80,6.60,70.70,246203,97542,5467,3364,414,4376.0,153.10,3.0,132.69,1.0,122.12,3.0,184.61,3.0,157.67,1.0,145.93,3.0
3523,Florida,Alachua County,12001,2012,7.30,6.90,7.60,70.00,247973,96544,5758,3416,392,4426.0,156.66,3.0,135.64,1.0,124.79,3.0,189.42,3.0,160.50,1.0,148.89,3.0
3524,Florida,Alachua County,12001,2013,8.10,7.50,8.60,68.60,249848,96043,5496,3799,372,4632.0,160.22,1.0,138.58,1.0,127.45,1.0,194.22,1.0,163.33,1.0,151.85,1.0
3525,Florida,Alachua County,12001,2014,8.10,7.70,8.60,69.70,251759,96137,5808,3705,365,4969.0,159.70,1.0,139.78,1.0,124.58,1.0,201.95,1.0,169.67,1.0,156.19,1.0
3526,Florida,Alachua County,12001,2015,7.40,6.80,8.00,74.00,254218,96703,6176,3656,503,4568.0,159.18,1.0,140.97,1.0,121.71,1.0,209.68,1.0,176.01,1.0,160.53,1.0
3527,Florida,Alachua County,12001,2016,6.50,5.70,7.30,74.00,256581,96336,6155,3718,555,4423.0,158.66,1.0,142.17,1.0,118.84,1.0,217.41,1.0,182.35,1.0,164.87,1.0
3528,Florida,Alachua County,12001,2017,6.20,5.60,6.70,70.20,259865,97485,6417,3802,531,4285.0,158.14,1.0,143.36,1.0,115.97,1.0,225.14,1.0,188.69,1.0,169.21,1.0


In [10]:
#finding the count of NaN values in each column
nan_counts2 = florida_df.isna().sum()

#Display the counts of NaN values for each column
print(nan_counts2)

State_Name              0
County_Name             0
County_FIPS_Code        0
StudyYear               0
UNR_20to64              0
FUNR_20to64             0
MUNR_20to64             0
FLFPR_20to64_Under6     0
TotalPop                0
Households              0
H_Under6_BothWork       0
H_Under6_FWork          0
H_Under6_MWork          0
H_Under6_SingleM        0
MCInfant               76
MCInfant_flag          76
MCToddler              76
MCToddler_flag         76
MCPreschool            76
MCPreschool_flag       76
_75CInfant             76
_75CInfant_flag        76
_75CToddler            76
_75CToddler_flag       76
_75CPreschool          76
_75CPreschool_flag     76
dtype: int64


### Observation: 
There are 76 rows out of 737 that contain all NaN values for all 12 of the last columns. These columns describe the childcare prices for each age group. This data is important to the story and these rows should evaluated on whether they should be dropped...

In [11]:
# Filter the DataFrame to show all rows with NaN values
rows_with_nan = florida_df[florida_df.isna().any(axis=1)]

# Display the filtered DataFrame
rows_with_nan

,State_Name,County_Name,County_FIPS_Code,StudyYear,UNR_20to64,FUNR_20to64,MUNR_20to64,FLFPR_20to64_Under6,TotalPop,Households,H_Under6_BothWork,H_Under6_FWork,H_Under6_MWork,H_Under6_SingleM,MCInfant,MCInfant_flag,MCToddler,MCToddler_flag,MCPreschool,MCPreschool_flag,_75CInfant,_75CInfant_flag,_75CToddler,_75CToddler_flag,_75CPreschool,_75CPreschool_flag
3529,Florida,Alachua County,12001,2018,5.7,4.9,6.5,69.1,263148,97048,7023,3659,462,4078.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3540,Florida,Baker County,12003,2018,5.5,4.8,6.1,77.4,27785,8625,1066,242,2,284.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3551,Florida,Bay County,12005,2018,5.8,5.5,6.1,71.4,182482,70199,4513,2895,200,4039.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3562,Florida,Bradford County,12007,2018,5.4,6.7,4.1,64.6,26979,8993,463,227,104,783.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3573,Florida,Brevard County,12009,2018,6.0,5.1,6.8,65.4,576808,228888,10496,7905,789,9182.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3584,Florida,Broward County,12011,2018,6.2,6.5,5.8,76.2,1909151,682088,48459,23735,1816,41616.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3595,Florida,Calhoun County,12013,2018,5.8,6.7,5.1,77.7,14444,4556,347,118,15,192.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3606,Florida,Charlotte County,12015,2018,7.1,7.3,7.0,70.8,176954,76150,1864,1444,178,1911.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3617,Florida,Citrus County,12017,2018,8.1,6.6,9.4,67.6,143087,62762,1554,1434,140,1842.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN
3628,Florida,Clay County,12019,2018,6.9,8.4,5.6,69.4,207291,73398,4864,3330,367,3461.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


### Observation: 
The 76 rows with 12 columns of NaN values are spread out throughout different counties in Florida. Therefore, not just one county's data will be affected if these rows are dropped. 

In [13]:
#drop rows with any NaN values, in place
florida_df.dropna(inplace=True)

#display the modified DataFrame
florida_df

,State_Name,County_Name,County_FIPS_Code,StudyYear,UNR_20to64,FUNR_20to64,MUNR_20to64,FLFPR_20to64_Under6,TotalPop,Households,H_Under6_BothWork,H_Under6_FWork,H_Under6_MWork,H_Under6_SingleM,MCInfant,MCInfant_flag,MCToddler,MCToddler_flag,MCPreschool,MCPreschool_flag,_75CInfant,_75CInfant_flag,_75CToddler,_75CToddler_flag,_75CPreschool,_75CPreschool_flag
3519,Florida,Alachua County,12001,2008,4.10,4.70,3.60,67.10,239046,96894,5618,3675,498,4499.0,137.65,3.0,115.54,1.0,106.36,3.0,165.00,3.0,138.50,1.0,127.50,3.0
3520,Florida,Alachua County,12001,2009,5.80,5.60,6.00,67.70,238990,96518,5236,4026,398,4258.0,145.99,3.0,126.80,1.0,116.79,3.0,175.00,3.0,152.00,1.0,140.00,3.0
3521,Florida,Alachua County,12001,2010,5.80,6.10,5.60,67.30,244247,97540,5128,3696,431,4298.0,149.55,3.0,129.75,1.0,119.46,3.0,179.81,3.0,154.83,1.0,142.96,3.0
3522,Florida,Alachua County,12001,2011,6.70,6.80,6.60,70.70,246203,97542,5467,3364,414,4376.0,153.10,3.0,132.69,1.0,122.12,3.0,184.61,3.0,157.67,1.0,145.93,3.0
3523,Florida,Alachua County,12001,2012,7.30,6.90,7.60,70.00,247973,96544,5758,3416,392,4426.0,156.66,3.0,135.64,1.0,124.79,3.0,189.42,3.0,160.50,1.0,148.89,3.0
3524,Florida,Alachua County,12001,2013,8.10,7.50,8.60,68.60,249848,96043,5496,3799,372,4632.0,160.22,1.0,138.58,1.0,127.45,1.0,194.22,1.0,163.33,1.0,151.85,1.0
3525,Florida,Alachua County,12001,2014,8.10,7.70,8.60,69.70,251759,96137,5808,3705,365,4969.0,159.70,1.0,139.78,1.0,124.58,1.0,201.95,1.0,169.67,1.0,156.19,1.0
3526,Florida,Alachua County,12001,2015,7.40,6.80,8.00,74.00,254218,96703,6176,3656,503,4568.0,159.18,1.0,140.97,1.0,121.71,1.0,209.68,1.0,176.01,1.0,160.53,1.0
3527,Florida,Alachua County,12001,2016,6.50,5.70,7.30,74.00,256581,96336,6155,3718,555,4423.0,158.66,1.0,142.17,1.0,118.84,1.0,217.41,1.0,182.35,1.0,164.87,1.0
3528,Florida,Alachua County,12001,2017,6.20,5.60,6.70,70.20,259865,97485,6417,3802,531,4285.0,158.14,1.0,143.36,1.0,115.97,1.0,225.14,1.0,188.69,1.0,169.21,1.0


In [14]:
#displaying summary statistics for the DataFrame
florida_df.describe()

,County_FIPS_Code,StudyYear,UNR_20to64,FUNR_20to64,MUNR_20to64,FLFPR_20to64_Under6,TotalPop,Households,H_Under6_BothWork,H_Under6_FWork,H_Under6_MWork,H_Under6_SingleM,MCInfant,MCInfant_flag,MCToddler,MCToddler_flag,MCPreschool,MCPreschool_flag,_75CInfant,_75CInfant_flag,_75CToddler,_75CToddler_flag,_75CPreschool,_75CPreschool_flag
count,661.000000,661.000000,661.000000,661.000000,661.000000,661.000000,6.610000e+02,661.000000,661.000000,661.000000,661.000000,661.000000,661.000000,661.000000,661.000000,661.0,661.000000,661.000000,661.000000,661.000000,661.000000,661.0,661.000000,661.000000
mean,12068.095310,2012.479576,9.261694,9.107156,9.414796,67.652027,2.884467e+05,109128.904690,6769.856278,4172.931921,369.027231,5799.098336,144.258442,1.928896,122.442995,1.0,110.568926,1.623298,161.526672,1.928896,138.105401,1.0,125.754539,1.623298
std,38.390912,2.873857,2.679307,2.907415,3.173691,7.748009,4.536514e+05,163870.097259,11538.636353,6625.078746,591.907463,10016.519045,26.369410,0.998224,21.963202,0.0,18.652365,0.927036,30.015592,0.998224,25.286994,0.0,22.756946,0.927036
min,12001.000000,2008.000000,2.450000,1.460000,1.340000,33.100000,7.723000e+03,1884.000000,86.000000,4.000000,0.000000,60.000000,77.550000,1.000000,64.630000,1.0,64.630000,1.000000,90.000000,1.000000,75.000000,1.0,75.000000,1.000000
25%,12037.000000,2010.000000,7.300000,7.100000,7.100000,64.100000,2.853700e+04,8824.000000,604.000000,376.000000,25.000000,672.000000,125.550000,1.000000,108.000000,1.0,97.340000,1.000000,139.690000,1.000000,120.000000,1.0,107.560000,1.000000
50%,12069.000000,2012.000000,9.100000,8.800000,9.100000,68.500000,1.167540e+05,50928.000000,1873.000000,1343.000000,139.000000,1832.000000,142.500000,1.000000,121.720000,1.0,110.160000,1.000000,156.860000,1.000000,137.000000,1.0,125.000000,1.000000
75%,12101.000000,2015.000000,10.900000,10.800000,11.200000,72.100000,3.188860e+05,122972.000000,6909.000000,4537.000000,397.000000,6199.000000,160.490000,3.000000,137.210000,1.0,123.050000,3.000000,179.830000,3.000000,154.070000,1.0,140.550000,3.000000
max,12133.000000,2017.000000,18.600000,20.200000,21.000000,95.800000,2.702602e+06,858289.000000,68611.000000,40837.000000,3915.000000,61036.000000,228.650000,3.000000,196.190000,1.0,160.560000,3.000000,273.670000,3.000000,214.030000,1.0,191.210000,3.000000


In [15]:
#getting data type info for the dataframe
florida_df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 661 entries, 3519 to 4254
Data columns (total 26 columns):
 #   Column               Non-Null Count  Dtype  
---  ------               --------------  -----  
 0   State_Name           661 non-null    object 
 1   County_Name          661 non-null    object 
 2   County_FIPS_Code     661 non-null    int64  
 3   StudyYear            661 non-null    int64  
 4   UNR_20to64           661 non-null    float64
 5   FUNR_20to64          661 non-null    float64
 6   MUNR_20to64          661 non-null    float64
 7   FLFPR_20to64_Under6  661 non-null    float64
 8   TotalPop             661 non-null    int64  
 9   Households           661 non-null    int64  
 10  H_Under6_BothWork    661 non-null    int64  
 11  H_Under6_FWork       661 non-null    int64  
 12  H_Under6_MWork       661 non-null    int64  
 13  H_Under6_SingleM     661 non-null    float64
 14  MCInfant             661 non-null    float64
 15  MCInfant_flag        661 non-null    floa

In [16]:
#UN-COMMENT WHEN READY TO SAVE TO FILE

#saving the filtered data (specific to my story) as an excel file. 
###florida_df.to_excel('florida_data.xlsx', index=False)